### PR TITLE
Masterbar: Add signup and login links when logged out

### PR DESCRIPTION
**Pull request by seear**
This adds links to the masterbar for any logged-out _Calypso_ page, e.g. https://wordpress.com/start or https://wordpress.com/design

**Before**
screen shot 2016-04-21 at 16 07 47
screen shot 2016-04-21 at 16 07 33

**After**
screen shot 2016-04-21 at 16 04 37
screen shot 2016-04-21 at 16 04 17

**To Test**
* Go to http://calypso.localhost:3000/devdocs/start
* The _Sign up_ link should go to /start
* The _Log in_ link should prompt for a login

NOTE: the _Sign up_ link will not work properly from the /design or /theme routes until #4864 is merged.

### CODE REVIEW

**Comment by lancewillett** [comment]
For history sake — could you add screenshots and description of the *current* behavior — what is this replacing now?

**Comment by lancewillett** [comment]
I see the before but it's not clear which view that is.

`https://wordpress.com/?apppromo`
 screen shot 2016-04-27 at 9 35 52 am

`wordpress.com`
screen shot 2016-04-27 at 9 35 56 am

`horizon.wordpress.com`

screen shot 2016-04-27 at 9 36 01 am

**Comment by seear** [comment]
> For history sake — could you add a better description of the current behavior — what is this replacing now?

@lancewillett: this is logged-out Calypso only, which isn't much, only these two I think:  https://wordpress.com/start or https://wordpress.com/design.

Current behaviour is no links other than the logo, which goes to wordpress.com.

**Comment by lancewillett** [comment]
Thanks! Here are two benchmarks for mobile view.

https://wordpress.com/start

screen shot 2016-04-27 at 9 58 25 am

https://wordpress.com/design

screen shot 2016-04-27 at 9 58 37 am

**Comment by lancewillett** [comment]
One note on wording: we should match current capitalization: 

`Sign Up` and `Log In` with title case.

**Comment by seear** [comment]
> we should match current capitalization

Good point, changed in ea66eb4

File: client/layout/masterbar/logged-out.jsx [hunk 1]
```diff
@@ -10,17 +10,23 @@ import Masterbar from './masterbar';
...
 		</Item>
+		<div className="masterbar__login-links">
+			<Item url={ config( 'signup_url' ) }>
+				{ translate( 'Sign Up', { context: 'Toolbar, must be shorter than ~12 chars' } ) }
```
Comment (ockham) [inline client/layout/masterbar/logged-out.jsx]: Super-nitpick here -- IIRC, `context` becomes a differentiating feature to the translation string, and should hence be sort of short, expressive, and easily reproducible by others. So I think it should be just something like `toolbar` here, while the note on length -- which is a nudge directed to the translator -- should go to `comment`. (Same on L26)

**Comment by ockham** [comment]
LGTM! Even works nicely with SSR'ed theme detail sheets!

**Comment by ockham** [comment]
(Don't forget to squash before merging tho -- happens to me way too often ;-)

**Pull request merged**